pxr/usdImaging/usdviewq/appController.py

```diff
@@ -2185,16 +2188,34 @@ def _findPrims(self, pattern, useRegex=True):
             pattern = pattern.lower()
             isMatch = lambda x: pattern in x.lower()
 
-        matches = [prim.GetPath() for prim
-                   in Usd.PrimRange.Stage(self._dataModel.stage,
-                                             self._displayPredicate)
-                   if isMatch(prim.GetName())]
+        matches = []
+        if self._dataModel.viewSettings.showPrimDisplayNames:
+            for prim in Usd.PrimRange.Stage(self._dataModel.stage, self._displayPredicate):
+                if prim.HasAuthoredDisplayName():
+                    if isMatch(prim.GetDisplayName()):
+                        matches.append(prim.GetPath())
+                elif isMatch(prim.GetName()):
+                    matches.append(prim.GetPath())
```

Comment:
What would you think about enhancing the `isMatch` lambda to take the prim and `useDisplayNames` as input ?  I think it would simplify the logic here, and reduce/eliminate code duplication with handling prototypes, below?

Reply:
I refactored the code into a true function (that still used the same lambdas underneath).  Trying to fit it all into a single lambda expression made it a little unreadable.